### PR TITLE
Added clap-rs and basic flag parsing

### PR DESCRIPTION
**Pull request by robbym**
Depends on https://github.com/ry/deno_third_party/pull/1
Any change suggestions would be appreciated.
```
$ deno
error: The following required arguments were not provided:
    <V8ARGS>...

USAGE:
    deno [FLAGS] <V8ARGS>...

For more information try --help
```
```
$ deno --help
deno 0.0.0
Ryan D. <ry@tinyclouds.org>
A secure TypeScript runtime built on V8

USAGE:
    deno [FLAGS] <V8ARGS>...

FLAGS:
    -d, --debug      Sets the log level to max
    -h, --help       Prints help information
    -V, --version    Prints version information

ARGS:
    <V8ARGS>...
```
```
$ deno --version
deno 0.0.0
```
```
$ deno --debug tests/002_hello.ts
Hello World
```

### CODE REVIEW

File: build_extra/rust/BUILD.gn [hunk 1]
```diff
@@ -69,3 +69,25 @@ rust_component("cfg_if") {
   source_root =
       "$crates/registry/src/github.com-1ecc6299db9ec823/cfg-if-0.1.4/src/lib.rs"
 }
+
+rust_component("clap") {
+  source_root =
+      "$crates/registry/src/github.com-1ecc6299db9ec823/clap-2.32.0/src/lib.rs"
+  extern = [ ":unicode_width", ":textwrap", ":bitflags" ]
+}
```
Comment (ry) [inline build_extra/rust/BUILD.gn]: > "Clap is a simple-to-use, efficient, and full-featured library"

I didn't explain this well in the issue, but I just want basic parsing in the Rust layer - and then pass the majority of argv off to JS where there will be further flag parsing. I think [getopts](https://github.com/rust-lang-nursery/getopts) would work for this?

File: src/main.rs [hunk 2]
```diff
@@ -86,11 +90,26 @@ impl Drop for Deno {
...
+    .arg(
+      Arg::with_name("debug")
+        .short("d")
+        .long("debug")
+        .help("Sets the log level to max"),
+    )
+    .get_matches();
+
+  if args.is_present("debug") {
+    log::set_max_level(log::LevelFilter::Debug);
```
Comment (ry) [inline src/main.rs]: Does this work for you? You should get more output.. maybe this isn't the right call?

Comment (robbym) [reply src/main.rs]: It doesn't output more regardless of the branch. Not sure. Haven't used the log crate unfortunately. Can look into it.

**Comment by robbym** [comment]
Switching to getopt crate.

**Pull request closed without merging**